### PR TITLE
Fix use of const ref in Abiv2 and other proto fuzzers.

### PR DESCRIPTION
**Pull request by bshastry**
Fixes issue found by nightly regression run.

### CODE REVIEW

File: test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp [hunk 1]
```diff
@@ -58,8 +58,9 @@ DEFINE_PROTO_FUZZER(Contract const& _contract)
 		// We target the default EVM which is the latest
 		langutil::EVMVersion version;
 		EVMHost hostContext(version, evmone);
-		string contractName = ":C";
-		CompilerInput cInput(version, {{"test.sol", contractSource}}, contractName, OptimiserSettings::minimal(), {});
+		string contractName = "C";
```
Comment (cameel) [inline test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp:61]: Why is the colon not needed here? I see that `contractName` is passed to `CompilerStack` and compiler stack identifies contracts with `<file name>:<contract name>` so even if `<file name>` is empty, the colon should still be there.

My assumption would be that some piece of code in `CompilerInput` or a related class is responsible concatenating the file name and contract name and inserts the colon but I do not see any such code (I did not look too deeply though).

File: test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp [hunk 2]
```diff
@@ -58,8 +58,9 @@ DEFINE_PROTO_FUZZER(Contract const& _contract)
 		// We target the default EVM which is the latest
 		langutil::EVMVersion version;
 		EVMHost hostContext(version, evmone);
-		string contractName = ":C";
```
Comment (cameel) [inline test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp:61]: I think you should update the docstring for `CompilerInput.contractName` to say how the value is supposed to look like. Especially whether it requires the colon or not.

File: test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp (hunk 1 above)
Comment (bshastry) [reply test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp:61]: I think this issue (not removing the colon) should have been done in an earlier PR where the utility class was refactored but overlooked.

I use contract name the same way as in here for example https://github.com/ethereum/solidity/blob/f188f3d9ed261223e16bcacff0d55c30e94500db/test/libsolidity/GasCosts.cpp#L197

File: test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp (hunk 2 above)
Comment (bshastry) [reply test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp:61]: Updated.

File: test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp (hunk 1 above)
Comment (cameel) [reply test/tools/ossfuzz/AbiV2IsabelleFuzzer.cpp:61]: So I'm curious why this works. I remember that when writing tests for `FunctionCallGraph` the colon was always there in the name...

But well, if it works I guess it must be ok.

**Review by cameel — APPROVED** [review]
I'm marking a approved but some tests fail. It might be unrelated to the PR though.

**Pull request merged**